aten/src/ATen/native/CPUFallback.cpp

```diff
@@ -0,0 +1,152 @@
+  // the temporary CPU output tensor that we created.
+  //
+  // Note [CPU Fallback Does Not Handle View Operators]
+  // Also note that we are incapable of handling immutable alises properly.
+  // Why?
+  // Schemas with an immutable alias'd tensor outputs correspond to view operators.
+  // For example, the `view_as` schema from native_functions.yaml:
+  // `view_as(Tensor(a) self, Tensor other) -> Tensor(a)`
+  // We can't handle these ops properly, because view ops are supposed to return
+  // a NEW tensor that shares the SAME storage as the original tensor.
+  // However, the new tensor that we created cannot share the same storage,
+  // since it lives on CPU and the original tensor lives on a different device.
+  // Because of that, we explicitly error out if someone attempts to call the
+  // CPU fallback on a view operator.
+  const auto& schema_returns = op.schema().returns();
+  const auto& num_returns = schema_returns.size();
+  auto returns = torch::jit::last(stack, num_returns);
+  const auto returns_begin = stack->size() - num_returns;
+
+  for (int64_t idx = 0; idx < returns.size(); ++idx) {
+    if (returns[idx].isTensor()) {
+      const auto& return_tens = returns[idx].toTensor();
+      if (return_tens.defined()) {
+        const auto& alias_info = schema_returns[idx].alias_info();
+        if (alias_info.has_value() && alias_info.value().isWrite()) {
+          // mutable alias case: move the input ivalue directly onto the stack
+          // in place of the existing cpu output tensor.
+          bool found_alias = false;
+          for (int64_t i = 0; i < tensor_args_indices.size(); ++i) {
+            auto input_tensor_idx = tensor_args_indices[i];
+            const auto& input_tensor = cpu_tensors[i];
+            const auto& input_alias_info = schema_args[input_tensor_idx].alias_info();
+            if (input_tensor.defined() && alias_info == input_alias_info) {
+              // We've found the original input tensor that aliases with the current output.
+              // Wrap it in an IValue and put it directly on the stack.
+              (*stack)[returns_begin + idx] = c10::IValue(tensor_args[i]);
+              found_alias = true;
+              break;
+            }
+          }
+          TORCH_CHECK(found_alias, "The operator ", op.schema().operator_name(), " appears to have invalid alias information. ",
+                      "Found a return tensor argument with a mismatched mutable alias: ", schema_returns[idx]);
+        } else if (alias_info.has_value() && !alias_info.value().isWrite()) {
+          // immutable alias (view) case: Fail here. If this operator is needed, the backend should provide a kernel for it.
+          // See Note [CPU Fallback Does Not Handle View Operators]
+          auto tgt_device = tensor_args[0].device();
+          TORCH_CHECK(false, "The operator ", op.schema().operator_name(), " appears to be a view operator, ",
```

Comment:
(is it just me or is GitHub not letting me see the original comment thread from the outdated changes)

@ailzhang it looks like XLA doesn't have full coverage of `torch.as_strided` either. The existing kernel [falls back to CPU in most cases.](https://github.com/pytorch/xla/blob/280348439b23fd7aef5c57c9b92298f8890bcc98/torch_xla/csrc/aten_xla_type.cpp#L655). If that just seems like a coverage gap, I can file an issue (and not block the PR on that getting fixed, just need to modify some xla tests).
```
>>> t = torch.ones(4, device=xm.xla_device())
>>> t2 = t.as_strided((2,), (2,))
>>> t2[0] = -1
>>> t
tensor([1., 1., 1., 1.], device='xla:0') # should have been mutated
>>> t2
tensor([-1.,  1.], device='xla:0')
>>>
```

Reply:
I will strongly prefer not to error out in this case, reason stated in https://github.com/pytorch/xla/issues/2964#issuecomment-847274777

Reply:
Warning instead of erroring to avoid being BC-breaking seems reasonable to me.

Reply:
Tested out the warn:
```
>>> import torch
>>> import torch_xla.core.xla_model as xm
>>> a = torch.ones(4, device=xm.xla_device())
>>> b = a.as_strided((2,), (2,))
__main__:1: UserWarning: 0The operator aten::as_strided appears to be a view operator, but it has no implementation for the backend "xla:0". View operators don't support falling back to run on the CPU, since the tensor's storage cannot be shared across devices. (Triggered internally at  ../aten/src/ATen/native/CPUFallback.cpp:142.)
>>>
```